### PR TITLE
Don't emit U symbols in symbols files

### PR DESCRIPTION
**Pull request by kripken**
We only use them to know what libraries provide, and not what they require.

(For requirements, we let a library say "lib X depends on lib Y", so it doesn't actually check undefs.)

### CODE REVIEW

**Comment by sbc100** [comment]
Wow I didn't know we didn't use this information.  There is no reason why would couldn't use it I guess...  but I'm trying to move away from these symbols files anyway.

**Comment by kripken** [comment]
Yeah, if we can move away that's even better.

Instead of using the undefined symbols atm, all we do is let an entire library depend on another in its definition. So libc++ depends on libc. That's probably enough anyhow.

File: tools/update_symbols.py [hunk 1]
```diff
@@ -92,9 +93,7 @@ def filter_and_sort(symbols):
 
   lines = ['# Auto-generated by tools/update_symbols.py.  DO NOT EDIT.']
   for typ, name in symbols:
-    if typ == 'U':
-      lines.append("         %s %s" % (typ, name))
-    else:
```
[the remark this answers is not in the document]
Comment (hujiajie) [reply tools/update_symbols.py]: Can we utilize the `-defined-only` option provided by llvm-nm?

[the remark this answers is not in the document]
Comment (kripken) [reply tools/update_symbols.py]: Good ideas, thanks @sbc100 and @hujiajie ! I used `-defined-only` and simplified all the code to assume that there is no `U` anywhere.

**Pull request merged**